### PR TITLE
Create-new-menu - release/v2.1

### PR DESCRIPTION
**Pull request by kowalowy1234**

### CODE REVIEW

File: billennium/libs/ui/src/drawer/drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
+import { mdiChevronLeft } from '@mdi/js';
+import Icon from '@mdi/react';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import styled, { keyframes } from 'styled-components';
+import { DrawerMenuItem } from '../navbar';
+import { Heading } from '../typography';
+import { logo } from './logo';
+
+interface DrawerProps {
```
Comment (ajronn) [inline billennium/libs/ui/src/drawer/drawer.tsx:10]: move to models.ts

File: billennium/apps/billespace/src/app/modules/menu/menu.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <>
+      <Navbar
+        mainItem={HomePage}
+        rightItem={Qr}
+        openDrawer={() => setOpen(true)}
+      />
+      <Drawer
+        links={MenuItems}
+        open={open}
```
Comment (ajronn) [inline billennium/apps/billespace/src/app/modules/menu/menu.tsx:20]: this logic should be in this component - parent should decide when children is rendering

File: billennium/libs/ui/src/drawer/drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  open,
+  links,
+  setAnimation,
+}: DrawerProps) => {
+  const [slideIn, setSlide] = useState(true);
+  const navigate = useNavigate();
+
+  const close = () => {
+    setSlide(false);
+    setTimeout(() => {
```
Comment (ajronn) [inline billennium/libs/ui/src/drawer/drawer.tsx:153]: this timeout is necessary?

File: billennium/libs/ui/src/navbar/menu-models.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+export type DrawerMenuItem = {
```
Comment (ajronn) [inline billennium/libs/ui/src/navbar/menu-models.ts:3]: exclude to models.ts file

File: billennium/libs/ui/src/navbar/navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import { mdiMenu } from '@mdi/js';
+import Icon from '@mdi/react';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+import { DrawerMenuItem } from './menu-models';
+
+export interface NavbarProps {
```
Comment (ajronn) [inline billennium/libs/ui/src/navbar/navbar.tsx:7]: same as above

File: billennium/libs/ui/src/drawer/drawer.tsx (hunk 3 above)
Comment (kowalowy1234) [reply billennium/libs/ui/src/drawer/drawer.tsx:153]: It's necessary to preserve slide in animation

**Pull request merged**